src/librustc_middle/ty/query/on_disk_cache.rs

```diff
@@ -370,14 +370,13 @@ impl<'sess> OnDiskCache<'sess> {
     {
         let pos = index.get(&dep_node_index).cloned()?;
 
-        // Initialize `cnum_map` using the value from the thread that finishes the closure first.
-        self.cnum_map.init_nonlocking_same(|| Self::compute_cnum_map(tcx, &self.prev_cnums[..]));
+        self.cnum_map.get_or_init(|| Self::compute_cnum_map(tcx, &self.prev_cnums[..]));
```

Comment:
You can `let cnum_map = self.cnum_map.get_or_init(` to avoid `.unwrap on line 379